uncertainty_tools/CMakeLists.txt

```diff
@@ -13,68 +13,97 @@
-file(GLOB_RECURSE headers */*.hpp */*.h)
-
-add_library(${PROJECT_NAME} src/${PROJECT_NAME}/${PROJECT_NAME}.cpp)
-target_link_libraries(${PROJECT_NAME})
-add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
-
-
-#############
-## Install ##
-#############
-
-## Mark executables and/or libraries for installation
- install(TARGETS ${PROJECT_NAME}
-   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
- )
-
-## Mark cpp header files for installation
- install(DIRECTORY include/${PROJECT_NAME}/
-   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-   FILES_MATCHING PATTERN "*.h"
-   PATTERN ".svn" EXCLUDE
- )
-
-##########
-## Test ##
-##########
-
-# SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
-catkin_add_gmock(${PROJECT_NAME}-test test/${PROJECT_NAME}/uncertainty_tools_test.cpp)
-if(TARGET ${PROJECT_NAME}-test)
-  target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
+find_package(ros_environment REQUIRED)
+set(ROS_VERSION $ENV{ROS_VERSION})
+
+if(${ROS_VERSION} EQUAL 1)
+ 
+  ## Find catkin macros and libraries
+  ## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+  ## is used, also find other catkin packages
+  find_package(catkin REQUIRED COMPONENTS)
+
+  catkin_package()
+
+  ###########
+  ## Build ##
+  ###########
+
+    ## Specify additional locations of header files
+    ## Your package locations should be listed before other locations
+    include_directories(
+      include
+      ${catkin_INCLUDE_DIRS}
+    )
+
+  add_library(${PROJECT_NAME} src/${PROJECT_NAME}/${PROJECT_NAME}.cpp)
+  target_link_libraries(${PROJECT_NAME})
+  add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
+
+
+  #############
+  ## Install ##
+  #############
+
+  ## Mark executables and/or libraries for installation
+  install(TARGETS ${PROJECT_NAME}
+    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  )
+
+  ## Mark cpp header files for installation
+  install(DIRECTORY include/${PROJECT_NAME}/
+    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN ".svn" EXCLUDE
+  )
+
+  ##########
+  ## Test ##
+  ##########
+
+  # SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+  catkin_add_gmock(${PROJECT_NAME}-test test/${PROJECT_NAME}/uncertainty_tools_test.cpp)
+  if(TARGET ${PROJECT_NAME}-test)
+    target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
+  endif()
+
+else() #ROS2
+
+  find_package(ament_cmake_auto REQUIRED)
+
+  include_directories(
+    include
+  )
+
+  add_library(${PROJECT_NAME}
+    src/${PROJECT_NAME}/${PROJECT_NAME}.cpp)
+
+  target_link_libraries(${PROJECT_NAME})
+
+  ament_export_include_directories(include)
+  ament_export_libraries(${PROJECT_NAME})
+
+  ament_auto_package(
+    INSTALL_TO_SHARE
```

Comment:
No need for this INSTALL_TO_SHARE command if your aren't listing any non-code files to install. Just
amenet_auto_package() should be sufficient